honeybee_3dm/face.py

```diff
@@ -0,0 +1,145 @@
+    """
+    hb_type = " "  # Variable holding a Honeybee face type
+    hb_face_module = " "  # Variable holding a Honeybee module name
+
+    if layer == "roof":
+        hb_type = face_types.roof_ceiling
+        hb_face_module = Face
+    if layer == "wall":
+        hb_type = face_types.wall
+        hb_face_module = Face
+    if layer == "floor":
+        hb_type = face_types.floor
+        hb_face_module = Face
+    if layer == "airwall":
+        hb_type = face_types.air_boundary
+        hb_face_module = Face
+    if layer == "shade":
+        hb_type = None
+        hb_face_module = Shade
+    if layer == "aperture":
+        hb_type = None
+        hb_face_module = Aperture
+    if layer == "door":
+        hb_type = None
+        hb_face_module = Door
+    if layer == "Default":
+        pass
+
+    return (hb_type, hb_face_module)
+
+
+# Look at Rhino layers and
+for layer in layer_names:
+    hb_face_type = layer_name_to_hb_objects(layer)[0]
+    hb_face_module = layer_name_to_hb_objects(layer)[1]
+    layer_name = layer
+
+    # Gathering planar geometries from the Rhino layer
+    rhino_faces = [object for object in file.Objects if object.Attributes.LayerIndex ==
+                   layer_dict[layer_name]]
+
+    # Creating face names
+    face_names = []
+    for count, geo in enumerate(rhino_faces):
+        # If there's a user given name, use it
+        # TODO: Quality check for user input name will have to be created here
+        if len(geo.Attributes.Name) > 0:
+            face_names.append(geo.Attributes.Name)
+        # Use auto generated name
+        else:
+            name = layer_name + str(uuid.uuid4())[:8]
+            face_names.append(name)
+
+    # for each rhino geometry gathered
+    for i in range(len(rhino_faces)):
+        face = rhino_faces[i].Geometry
+        # Converting the Rhino3dm geometry into a Ladybug Face3D object
+        lb_face = to_face3d(face)
+        for j in range(len(lb_face)):
+            # Converting Ladybug Face3D into Honeybee Face
+            if hb_face_module == Face:
+                hb_face = hb_face_module(clean_and_id_string('{}_{}'.format(
+                    face_names[i], i)), lb_face[j], hb_face_type)
+            if hb_face_module == Shade:
+                hb_face = hb_face_module(clean_and_id_string(
+                    '{}_{}'.format(face_names[i], i)), lb_face[j])
+            if hb_face_module == Aperture:
+                hb_face = hb_face_module(clean_and_id_string(
+                    '{}_{}'.format(face_names[i], i)), lb_face[j])
+            if hb_face_module == Door:
+                hb_face = hb_face_module(clean_and_id_string(
+                    '{}_{}'.format(face_names[i], i)), lb_face[j])
+            # Assigning a name to the Honeybee Face
+            hb_face.display_name = '{}_{}'.format(face_names[i], i)
+            hb_faces.append(hb_face)
+
+# Write the Honeybee faces in a JSON file
+publish_json(hb_faces)
```

Comment:
This will probably change when we refactor this module to be a number of re-usable functions.

Reply:
Revised